pallets/marketplace/src/lib.rs

```diff
@@ -357,6 +392,54 @@ pub mod pallet {
             // removed fulfilled job from assigned jobs
             <StoredJobAssignment<T>>::remove(&who, &job_id);
 
+            // strips away the asset amount
+            let reward_asset: <T as Config>::AssetId = extra
+                .reward
+                .try_get_asset_id()
+                .map_err(|_| Error::<T>::JobRegistrationUnsupportedReward)?
+                .into();
+
+            let reward_amount: T::AssetAmount = extra
+                .reward
+                .try_get_amount()
+                .map_err(|_| Error::<T>::JobRegistrationUnsupportedReward)?
+                .into();
+
+            let avg_job_reward = <StoredAvgJobReward<T>>::get(&reward_asset).unwrap_or_default();
+            let total_jobs_assigned =
```

Comment:
Should it be default if not yet in map? Or rather ensure it is 0 by using 0.into()

Reply:
done in 9163ba3fcc510ebb04b4ec52d5456b9566943a19